src/backend/cuda/jit.cpp

```diff
@@ -72,12 +85,11 @@ static string getKernelString(const string funcName,
     const std::string includeFileStr(jit_cuh, jit_cuh_len);
 
     const std::string paramTStr = R"JIT(
-template<typename T>
 struct Param
 {
-  T *ptr;
   dim_t dims[4];
   dim_t strides[4];
+  void *ptr;
 };
 )JIT";
```

Comment:
I am doing this so that I can avoid adding types to Params since these param objects are really just used to store the stride and the dimensions of the inputs. I should probably make another type for this but it doesn't really add much.

Reply:
It would be just more consistent with how regular kernels compiled by nvrtc will also look like when traced. And one less type to manager in multiple locations.